programs/local/LocalServer.cpp

```diff
@@ -414,7 +414,13 @@ try
 {
     UseSSL use_ssl;
     ThreadStatus thread_status;
-    signal(SIGINT, signalHandler);
+
+    struct sigaction new_act;
+    struct sigaction old_act;
+    new_act.sa_handler = interruptSignalHandler;
```

Comment:
Don't we need to clear `sa_flags` and `sa_mask`?

Reply:
Sorry, I mean `sa_sigaction`, not `sa_mask`

Reply:
No, don't need, because sa_sigaction and sa_handler are a union in some architectures.
sa_flags and sa_mask were already cleared in the next lines.

Reply:
No, don't need, because sa_sigaction and sa_handler are a union in some architectures.
sa_flags and sa_mask were already cleared in the next lines.

Reply:
> sa_sigaction and sa_handler are a union in some architectures

`man 3 sigaction` says that "The storage occupied by sa_handler and sa_sigaction **may overlap**", but it may not. Also `sigaction` structure may contain some extra fields. We can clear all unused fields by using `memset`, see `addSignalHandler(...)` in `BaseDaemon.cpp`